fn_mxtoolbox/fn_mxtoolbox/components/fn_mxtoolbox.py

```diff
@@ -0,0 +1,61 @@
+# (c) Copyright IBM Corp. 2010, 2018. All Rights Reserved.
+# -*- coding: utf-8 -*-
+# pragma pylint: disable=unused-argument, no-self-use
+
+"""Function implementation"""
+
+import logging
+import requests
+from resilient_circuits import ResilientComponent, function, handler, StatusMessage, FunctionResult, FunctionError
+from fn_mxtoolbox.lib.resilient_common import validate_fields
+
+HEADERS = {'content-type': 'application/json'}
+class FunctionComponent(ResilientComponent):
+    """Component that implements Resilient function 'fn_mxtoolbox"""
+
+    def __init__(self, opts):
+        """constructor provides access to the configuration options"""
+        super(FunctionComponent, self).__init__(opts)
+        self.options = opts.get("fn_mxtoolbox", {})
+
+    @handler("reload")
+    def _reload(self, event, opts):
+        """Configuration options have changed, save new values"""
+        self.options = opts.get("fn_mxtoolbox", {})
+        self._init()
+
+    def _init(self):
+        validate_fields(['api_token','mx_command','mx_argument'], self.options)
```

Comment:
I think this validation will fail. api_token is correct as it comes from the app.config file, but mx_command and mx_argument will come through the function invocation from resilient.